packages/core/src/stream/aisdk/v5/compat/prepare-tools.ts

```diff
@@ -59,7 +60,9 @@ export function prepareToolsAndToolChoice<TOOLS extends Record<string, Tool>>({
                 type: 'function' as const,
                 name,
                 description: sdkTool.description,
-                inputSchema: asSchema(sdkTool.inputSchema).jsonSchema,
+                inputSchema: isJsonSchema(sdkTool.inputSchema)
```

Comment:
This isn't related to the fix, but I have a feeling this will be an issue too for sending client tool schemas through MastraClient as well